app/ante/eth.go

```diff
@@ -45,12 +45,8 @@ func NewEthSigVerificationDecorator(ek EVMKeeper) EthSigVerificationDecorator {
 
 // AnteHandle validates checks that the registered chain id is the same as the one on the message, and
 // that the signer address matches the one defined on the message.
+// It's not skipped for RecheckTx, because the `From` address is critical for other ante handler to work.
```

Comment:
I'd add more details to this comment summarizing the finding that you wrote on the chat

Reply:
done, added a new sentence.